### DISCUSSION TITLE
Can MediaMTX use a Nvidia GPU/Cuda/NVENC to process ingest video?

### DISCUSSION THREAD

**Discussion by Majestic7979** [bluenviron/mediamtx#2062] [Questions]
I have the Nvidia Docker toolkit installed in my Debian 12 server, as well as the confirmation that any container created in the CLI with the following commands can see the GPU using nvdia-smi

```
--runtime=nvidia -e RUN_OPTS="--noacl" --device /dev/dri/renderD128 --device /dev/bus/usb:/dev/bus/usb -v /dev/dri:/dev/dri --gpus=all --cap-add=CAP_PERFMON -e NVIDIA_VISIBLE_DEVICES=all -e NVIDIA_DRIVER_CAPABILITIES=compute,video,utility,compat32,graphics,display
```

Does MediaMTX ffmpeg have the ability to use cuda/nvenc? If not can this please be added? Running four 4K cameras and the fps isn't as smooth using the CPU because the server is running multiple applications not just MediaMTX. It has a GTX 1080 which is by any means a very powerful GPU for decoding/encoding, it's a waste to not use it.

**Comment by aler9** [comment]
Hello, the Docker image of the server is based on Alpine Linux, whose FFmpeg is not compiled with nvenc support. Even Ubuntu's FFmpeg is not compiled with nvenc support.

In order to use nvenc you can either run the undockerized server version or build a custom image based on `nvidia/cuda`, in which you have to put a nvenc-capable FFmpeg and the server executable.

**Reply by Majestic7979** [reply]
Could you do that instead? Like Frigate NVR project has done? They managed to bundle go2rtc with nvenc support, but I don't want to try and end up breaking functionality. Go2rtc is not working for me. Your application however works perfectly.

[a comment by an automated account is not shown]
